cartoframes/data/observatory/catalog/dataset.py

```diff
@@ -400,9 +400,11 @@ def to_csv(self, file_path, credentials=None, limit=None, order_by=None, sql_que
         """
         _credentials = get_credentials(credentials)
 
-        if not self._is_subscribed(_credentials):
-            raise DOError('You are not subscribed to this Dataset yet. '
-                          'Please, use the subscribe method first.')
+        if not self.is_public_data:
+            _subscribed_ids = subscriptions.get_subscription_ids(_credentials, DATASET_TYPE)
+            if self.id not in _subscribed_ids:
+                raise DOError('You are not subscribed to this Dataset yet. '
+                              'Please, use the subscribe method first.')
```

Comment:
This could be moved to a `self.check_subscription()` method that could be in the `CatalogEntity` class, and that method would raise the `DOError`.

If the `CatalogEntity` is not a good fit for this new method (other non subscriptable entities would extend from it), another option could be creating a new `Subscription` class, put the method there and make `Dataset` and `Geography` extend from both `CatalogEntity` and `Subscription`